tools/make-v8.sh

```diff
@@ -37,6 +40,6 @@ if [ "$ARCH" = "s390x" ] || [ "$ARCH" = "ppc64le" ]; then
   ninja -v -C "out.gn/$BUILD_ARCH_TYPE" d8 cctest inspector-test
 else
   # shellcheck disable=SC2086
-  PATH=~/_depot_tools:$PATH tools/dev/v8gen.py "$BUILD_ARCH_TYPE" --no-goma $V8_BUILD_OPTIONS
-  PATH=~/_depot_tools:$PATH ninja -C "out.gn/$BUILD_ARCH_TYPE/" d8 cctest inspector-test
+  PATH=~/_depot_tools:"$DEPOT_TOOLS_DIR":$PATH tools/dev/v8gen.py "$BUILD_ARCH_TYPE" --no-goma $V8_BUILD_OPTIONS
+  PATH=~/_depot_tools:"$DEPOT_TOOLS_DIR":$PATH ninja -C "out.gn/$BUILD_ARCH_TYPE/" d8 cctest inspector-test
```

Comment:
**Question for reviewers:** Can I remove the part where `~/_depot_tools` is added to the `PATH` in these lines? I believe it is redundant and doesn't actually seem to be used.

<details>

I believe the `~/_depot_tools` dir added to the `PATH` here is arbitrary and not useful, since in my local testing this dir is not created.

(It would be a coincidence, or an undocumented expectation, that the user or CI instance already manually downloaded `depot_tools` to `~/_depot_tools`. That seems entirely redundant when a copy of `depot_tools` is downloaded to `deps/v8/_depot_tools` at the top of this very script.)

Given the redundancy/arbitrariness/undocumentedness of `~/_depot_tools`, I would like to stop adding that to the `PATH`, in favor of _only_ adding  `deps/v8/_depot_tools` to PATH here.

(I didn't do that without asking, as leaving the existing path seemed more conservative, and in case it has some implication I can't be aware of for CI, since I don't know how the CI is set up. Maybe `~/_depot_tools` exists in CI? I am actually unclear how this passes in CI before my PR, unless CI already has `ninja` on the PATH somewhere, or has `~/_depot_tools`.)

</details>

Reply:
I just ran `make test-v8` all the way through again to confirm, starting from a clean clone of this repository. Same result. It cloned `depot_tools` to `deps/v8/_depot_tools`... I had no files/folders at `~/_depot_tools` at the end.

So that's why I think it makes sense to add `deps/v8/_depot_tools` to the PATH and stop adding `~/_depot_tools`. The former is created during this script, the latter is not.

Reply:
Yeah, maybe there was a historic reason but I can't see a reason for it in the current CI. 

The CI job does this before running the makefile:
```
# setup depot tools
git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git depot_tools
export PATH=$WORKSPACE/depot_tools:$PATH
```
but that isn't `~/_depot_tools`. 

From the V8 run: 
https://ci.nodejs.org/job/node-test-commit-v8-linux/3949/nodes=benchmark-ubuntu1604-intel-64,v8test=v8test/consoleFull
we end up with:
```
20:58:00 + PATH=/home/iojs/_depot_tools:/home/iojs/build/workspace/node-test-commit-v8-linux/nodes/benchmark-ubuntu1604-intel-64/v8test/v8test/deps/v8/_depot_tools:/home/iojs/build/workspace/node-test-commit-v8-linux/nodes/benchmark-ubuntu1604-intel-64/v8test/v8test/depot_tools:/home/iojs/nghttp2/src:/home/iojs/wrk:/usr/lib/ccache:/usr/lib64/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin tools/dev/v8gen.py x64.release --no-goma
20:58:00 + PATH=/home/iojs/_depot_tools:/home/iojs/build/workspace/node-test-commit-v8-linux/nodes/benchmark-ubuntu1604-intel-64/v8test/v8test/deps/v8/_depot_tools:/home/iojs/build/workspace/node-test-commit-v8-linux/nodes/benchmark-ubuntu1604-intel-64/v8test/v8test/depot_tools:/home/iojs/nghttp2/src:/home/iojs/wrk:/usr/lib/ccache:/usr/lib64/ccache:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin ninja -C out.gn/x64.release/ d8 cctest inspector-test
```
i.e.
1.  a non-existent dir (`~/_depot_tools`) 
1. depot_tools pulled down by the `tools/node/fetch_deps.py` script (`$WORKSPACE/deps/v8/_depot_tools`) added to the PATH in this PR
1. depot_tools git cloned by the CI job into `$WORKSPACE/depot_tools`

So the CI (for x64) was picking up `ninja` from `depot_tools` that was git cloned at the start of the job (and not the one later fetched by the `fetch_deps.py` script).

cc @nodejs/v8-update

Reply:
Trying to clean up and simplify where possible, I think modifying PATH may be unnecessary in general:

On my machine, this line that runs `v8gen.py` (right above the line invoking `ninja`) doesn't need anything added to the PATH on my machine:
https://github.com/nodejs/node/blob/7162e686b18d22b4385fa5c04274fb04dbd810c7/tools/make-v8.sh#L40

For me locally, it could be simplified to just:
`tools/dev/v8gen.py "$BUILD_ARCH_TYPE" --no-goma $V8_BUILD_OPTIONS`

The next line also works for me without anything added to the PATH if executing `deps/v8/_depot_tools/ninja` directly:

`"$DEPOT_TOOLS_DIR"/ninja -C "out.gn/$BUILD_ARCH_TYPE/" d8 cctest inspector-test`

So I believe the lines could be simplified (don't modify the PATH) to:

```
tools/dev/v8gen.py "$BUILD_ARCH_TYPE" --no-goma $V8_BUILD_OPTIONS
"$DEPOT_TOOLS_DIR"/ninja -C "out.gn/$BUILD_ARCH_TYPE/" d8 cctest inspector-test
```
Not sure if that would wok on CI or not.

Reply:
Lots of previous discussion in this PR, if more historical context is needed: https://github.com/nodejs/node/pull/19201

Particularly, the PATH modification and `~/_depot_tools` seem to have been copy-pasted from snippets suggested by a Google employee during some troubleshooting.

https://github.com/nodejs/node/pull/19201#issuecomment-371266836 then https://github.com/nodejs/node/pull/19201#issuecomment-372695079.

Reply:
In terms of non-x64 architectures we do test PPC64 and s390x and depot_tools is missing binaries for those architectures AFAIK (@miladfarca probably has a better idea on where we stand there) which is why the script has a different if branch that calls `gn` directly and not via `v8gen.py`.

Reply:
That's right depot_tools doesn't include gn and ninja binaries for PPC and s390. They need to be built/compiled from source and used directly.

Reply:
Thnks for the information.

As this PR does not modify the PPC or s390, I won't worry about those. I tried the Windows workflow to run V8 tests, and it seems it has been broken for some time, so I also won't worry about that.

<details><summary>Details of how the Windows <code>test-v8</code> workflow is broken (click to expand)</summary>

- The V8 tests on Windows are apparently meant to be run as `vcbuild.bat test-v8`.
- This runs `tools\test-v8.bat` (proof: [snippet](https://github.com/nodejs/node/blob/v16.0.0/vcbuild.bat#L630-L632))
  - This file was committed once in back in 2017 and never updated again (https://github.com/nodejs/node/commits/v16.0.0/tools/test-v8.bat)
  - This calls out to `tools\make-v8.sh` (proof: [snippet](https://github.com/nodejs/node/blob/v16.0.0/tools/test-v8.bat#L9-L10)), which has been substantially updated since 2017 and does not seem compatible with `tools\test-v8.sh` anymore.
- As it stands, I get some Python errors from some scripts in `depot_tools` during `vcbuild test-v8`.

</details>

That leaves x64 and arm64 (Linux and macOS), and armv7 (Linux) to possibly consider under Tier 1 or Tier 2 support from here: https://github.com/nodejs/node/blob/master/BUILDING.md#platform-list ... (Although it appears only a subset of those have been regularly running the V8 tests in CI? So de-facto we could continue only worrying about those platforms which are being run in CI?)

I only have access to an x64 macOS and x64 Linux machine, so that's what I can test. I'm pretty comfortable that the PR as-is works with both of my machines.

(I think it could be simplified as I mentioned above, but I'm happy to stop going to down that rabbit hole for now if it means addressing my original concern about the `ninja` dependency getting found during the script. Perhaps I should wait for another PR to provide evidence the simplified approach is good?)

Reply:
@DeeDeeG Thanks for digging into the history. If you want to make changes we can always start another CI run. I think as long as the CI is happy we can probably land if no-one objects.